staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go

```diff
@@ -0,0 +1,271 @@
+	apiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/server/healthz"
+	kubeinformers "k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	"k8s.io/pod-security-admission/admission"
+	admissionapi "k8s.io/pod-security-admission/admission/api"
+	podsecurityconfigloader "k8s.io/pod-security-admission/admission/api/load"
+	"k8s.io/pod-security-admission/cmd/webhook/server/options"
+	"k8s.io/pod-security-admission/policy"
+)
+
+// NewSchedulerCommand creates a *cobra.Command object with default parameters and registryOptions
+func NewServerCommand() *cobra.Command {
+	opts := options.NewOptions()
+
+	cmd := &cobra.Command{
+		Use: "podsecurity-webhook",
+		Long: `The PodSecurity webhook is a standalone webhook server implementing the Pod
+Security Standards.`,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runServer(cmd.Context(), opts)
+		},
+		Args: cobra.NoArgs,
+	}
+	opts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func runServer(ctx context.Context, opts *options.Options) error {
+	config, err := LoadConfig(opts)
+	if err != nil {
+		return err
+	}
+	server, err := Setup(config)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go func() {
+		stopCh := apiserver.SetupSignalHandler()
+		<-stopCh
+		cancel()
+	}()
+
+	return server.Start(ctx)
+}
+
+type Server struct {
+	secureServing   *apiserver.SecureServingInfo
+	insecureServing *apiserver.DeprecatedInsecureServingInfo
+
+	informerFactory kubeinformers.SharedInformerFactory
+
+	delegate *admission.Admission
+}
+
+func (s *Server) Start(ctx context.Context) error {
+	s.informerFactory.Start(ctx.Done())
+
+	mux := http.NewServeMux()
+	healthz.InstallHandler(mux, healthz.PingHealthz)
+	healthz.InstallReadyzHandler(mux, healthz.NewInformerSyncHealthz(s.informerFactory))
+	// The webhook is stateless, so it's safe to expose everything on the insecure port for
+	// debugging or proxy purposes. The API server will not connect to an http webhook.
+	mux.HandleFunc("/", s.HandleValidate)
+
+	if s.insecureServing != nil {
+		if err := s.insecureServing.Serve(mux, 0, ctx.Done()); err != nil {
+			return fmt.Errorf("failed to start insecure server: %w", err)
+		}
+	}
+
+	var shutdownCh <-chan struct{}
+	if s.secureServing != nil {
+		var err error
+		shutdownCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
+		if err != nil {
+			return fmt.Errorf("failed to start secure server: %w", err)
+		}
+	}
+
+	// Wait for graceful shutdown.
+	<-shutdownCh
+
+	return nil
+}
+
+func (s *Server) HandleValidate(w http.ResponseWriter, r *http.Request) {
+	defer utilruntime.HandleCrash(func(_ interface{}) {
+		// Assume the crash happened before the response was written.
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+	})
+
+	var (
+		body []byte
+		err  error
+		ctx  = r.Context()
```

Comment:
if there's a "timeout" query parameter, parse it into a duration and shorten the context timeout to that

that is is what the API server sends to the webhook to inform it of the timeout (shorter of the API server request timeout and the configured admission webhook timeout)